File: fastapi_asyncpg/backend/schema.py
from typing import List, Optional
from pydantic import BaseModel, NonNegativeInt, conint


class Version(BaseModel):
    version: str


class FibonacciResponse(BaseModel):
    result: conint(gt=0)


class ContactType(BaseModel):
    type_id: NonNegativeInt
    type_name: str


class Contact(BaseModel):
    contact_id: NonNegativeInt
    first_name: str
    last_name: str
    contact_type: ContactType
    phone_number: Optional[List[str]]
    email: Optional[List[str]]


class ContactTypesResponse(BaseModel):
    count: NonNegativeInt
    results: List[ContactType]


class ContactsResponse(BaseModel):
    count: NonNegativeInt
    results: List[Contact]

File: golang-bun/pkg/schema/schema.go
package schema

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

type Version struct {
	Version string `json:"version"`
}

type FibonacciParams struct {
	Number uint `uri:"number" binding:"required,gte=0,lte=40"`
}

type FibonacciResponse struct {
	Result uint `json:"result"`
}

type ContactType struct {
	ID   uint   `json:"type_id"`
	Name string `json:"type_name"`
}

type PhoneNumber string

type Email string

type Contact struct {
	ID          uint          `json:"contact_id"`
	FirstName   string        `json:"first_name"`
	LastName    string        `json:"last_name"`
	ContactType ContactType   `json:"contact_type"`
	PhoneNumber []PhoneNumber `json:"phone_number"`
	Email       []Email       `json:"email"`
}

type ContactParams struct {
	ContactID uint `uri:"contact_id" binding:"required,gt=0"`
}

type ContactTypesResponse struct {
	Count   uint          `json:"count"`
	Results []ContactType `json:"results"`
}

type ContactsParams struct {
	Limit  uint   `json:"limit" default:"100" binding:"lte=2000"`
	Offset uint   `json:"offset"`
	Starts string `json:"starts"`
}

type ContactsResponse struct {
	Count   uint      `json:"count"`
	Results []Contact `json:"result"`
}

func EncodeError(c *gin.Context, err error) {
	c.JSON(http.StatusBadRequest, gin.H{
		"error":   "fubar",
		"message": "fubar",
	})
}
